app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -17,10 +17,13 @@ def find_or_create_operating_system(distribution)
           major, minor = distribution.version.split('.')
           minor ||= '' # treat minor versions as empty string to not confuse with nil
 
-          os = ::Redhat.where(:name => os_name, :major => major, :minor => minor).first
-          os = create_operating_system(os_name, major, minor) unless os
+          create_os = lambda { ::Redhat.where(:name => os_name, :major => major, :minor => minor).first_or_create! }
 
-          return os
+          begin
+            create_os.call
+          rescue ActiveRecord::RecordInvalid
+            create_os.call
```

Comment:
So why call this again? Reading the notes on this, if the exception is thrown, that implies that the record already exists so calling it again would just simply find it but we don't appear to be needing to capture the return from the find itself.